pFUnittests/test_slamastroconv.pf

```diff
@@ -0,0 +1,233 @@
+!!
+!> @author  Arkene Chouchene (AC)
+!!
+!> @date    <ul>
+!!            <li>20.05.2023 (initial design)</li>
+!!          </ul>
+!!
+!! @details Program for testing slam_astro_conversion functions and subroutines 
+!! Sources and Examples are mentioned above every test
+!------------------------------------------------------------------------
+module test_slamastroconv
+    use funit
+    use slam_astro_conversions
+    use slam_time
+    use iso_fortran_env
+    use slam_error_handling
+    use slam_strings, only: toLowercase, toUppercase
+    use slam_types
+    use slam_math, only: pi, halfPi, twoPi, rad2deg, deg2rad
+        
+    implicit none
+        
+    contains
+          
+   @test
+   subroutine test_rv2coe_long()
+
+        ! Tests the rv2coe_long subroutine
+        ! Reference Data : Astropy
+        ! Declare variables
+        real(dp), dimension(3) :: R, V
+        real(dp) :: P, A, Ecc, Incl, Omega, Argp, Nu, M, ArgLat, TrueLon, LonPer
+        integer :: otype
+        real(dp) :: expected_P, expected_A, expected_Ecc, expected_Incl, expected_Omega, expected_Argp
+        real(dp) :: expected_Nu, expected_M, expected_ArgLat, expected_TrueLon, expected_LonPer
+
+        ! Initialize the input vectors
+        R = (/-3485.799126707284_dp, -5898.652976745232_dp, 835.9701786284777_dp/)
+        V = (/-1.3525457950562447_dp, -0.2804534841971075_dp, -7.4721873681232385_dp/)
+
+        ! Call the rv2coe_long subroutine
+        call rv2coe_long(R, V, P, A, Ecc, Incl, Omega, Argp, Nu, M, ArgLat, TrueLon, LonPer, otype)
+
+        ! Define the expected values
+        expected_P = 6901.6785750976933_dp
+        expected_A = 6901.7162837732567_dp
+        expected_Ecc = 0.23374487008203386E-2_dp
+        expected_Incl = 1.7046673456441539_dp
+        expected_Omega = 1.0206263838483220_dp
+        expected_Argp = 1.4007653593235663_dp
+        expected_Nu = 1.6183154619738946_dp
+        expected_M = 1.6136454570550320_dp
+        expected_ArgLat = 999999.09999999998_dp
+        expected_TrueLon = 999999.09999999998_dp
+        expected_LonPer = 999999.09999999998_dp
+
+        ! Check if the computed values match the expected values
+        @assertRelativelyEqual(P, expected_P, tolerance=1.0e-5_dp)
+        @assertRelativelyEqual(A, expected_A, tolerance=1.0e-5_dp)
+        @assertRelativelyEqual(Ecc, expected_Ecc, tolerance=1.0e-5_dp)
+        @assertRelativelyEqual(Incl, expected_Incl, tolerance=1.0e-5_dp)
+        @assertRelativelyEqual(Omega, expected_Omega, tolerance=1.0e-5_dp)
+        @assertRelativelyEqual(Argp, expected_Argp, tolerance=1.0e-5_dp)
+        @assertRelativelyEqual(Nu, expected_Nu, tolerance=1.0e-5_dp)
+        @assertRelativelyEqual(M, expected_M, tolerance=1.0e-5_dp)
+        @assertRelativelyEqual(ArgLat, expected_ArgLat, tolerance=1.0e-5_dp)
+        @assertRelativelyEqual(TrueLon, expected_TrueLon, tolerance=1.0e-5_dp)
+        @assertRelativelyEqual(LonPer, expected_LonPer, tolerance=1.0e-5_dp)
+
```

Comment:
The expected and actual values are switched, this is also the case for all other assertions. Not a huge deal but may be confusing in case of failing tests